tests/unit/evaluation/CrossValidation_multithread_unittest.cc

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (w) 2016 Saurabh Mahindre
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+
+#include <shogun/base/init.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/labels/BinaryLabels.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/classifier/svm/LibSVM.h>
+#include <shogun/evaluation/CrossValidation.h>
+#include <shogun/evaluation/StratifiedCrossValidationSplitting.h>
+#include <shogun/evaluation/ContingencyTableEvaluation.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+
+void generate_data(SGMatrix<float64_t>& mat, SGVector<float64_t> &lab)
+{
+	int32_t num=lab.size();
+	SG_SPRINT("%d", num);
+
+	for (index_t i=0; i<num; i++)
+	{
+		mat(0,i)=i<num/2 ? 0 : 100;
+		mat(1,i)=i;
+	}
+
+	for (index_t i=0; i<num; ++i)
+		lab.vector[i]=i<num/2 ? -1.0 : 1.0;
+
+}
+
+TEST(CrossValidation_multithread, LibSVM_unlocked)
+{
+	int32_t num=100;
+	SGMatrix<float64_t> mat(2, num);
+
+	/* training labels +/- 1 for each cluster */
+	SGVector<float64_t> lab(num);
+
+	/*create simple linearly separable data*/
+	generate_data(mat, lab);
```

Comment:
this method could return a std::pair<CFeatures*, CLabels*> with newly allocated data, so this code gets slimmer